File: test/Test_Mayonaise.cpp
#include "../src/Mayonaise.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(MayonaiseTest, AllFunctionalityTest){
    Mayonaise testMayonaise(50);
    LinkedList<FarmProductType> MayonaiseRecipe = testMayonaise.getResep();
    ASSERT_EQ(EGG, MayonaiseRecipe.get(0));
    ASSERT_EQ(MILK, MayonaiseRecipe.get(1));
    ASSERT_EQ(EGG, testMayonaise.getResepwithIdx(0));
    ASSERT_EQ(MILK, testMayonaise.getResepwithIdx(1));
    ASSERT_EQ(50, testMayonaise.getPrice());
    ASSERT_EQ(MAYONAISE, testMayonaise.getObjectType());
    ASSERT_EQ(SIDEPRODUCT, testMayonaise.getProductType());
}

File: Java/src/test/java/engisubs/gameobject/product/sideproduct/TestMayonaise.java
package engisubs.gameobject.product.sideproduct;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import engisubs.gameobject.GameObject.FarmProductType;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;

public class TestMayonaise{
    private Mayonaise test;

    @Before
    public void setUpObject() {
        test = new Mayonaise(80);
    }
    @Test
    public void instantiationTest(){
        Assert.assertEquals(80, test.getPrice());
        Assert.assertEquals(GameObjectType.MAYONAISE, test.getGameObjectType());
        Assert.assertEquals(ProductType.SIDEPRODUCT, test.getProductType());
        Assert.assertEquals("Mayo : The Nice Sauce for a Nice Food~~", test.getProductName());
    }
    @Test
    public void resepTest(){
        Mayonaise.InitResep();
        Assert.assertArrayEquals(new FarmProductType[]{FarmProductType.EGG, FarmProductType.MILK}, Mayonaise.getResep().toArray());
        Assert.assertEquals(null, Mayonaise.getResepwithIdx(3));
    }
}